File: src/test/java/pl/training/workshop/m6/s03_typecode/S03SolutionTest.java
package pl.training.workshop.m6.s03_typecode;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;

import org.junit.jupiter.api.Test;

import pl.training.workshop.m6.s03_typecode.step3.Format;
import pl.training.workshop.m6.s03_typecode.step3.FormatCodes;

/** Granica trwałości: w bazie zostaje stabilny kod, nie ordinal ani nazwa stałej. */
final class S03SolutionTest {
    @Test
    void persistentCodeIsNotTheOrdinal() {
        assertNotEquals(Format.IMAX.ordinal(), FormatCodes.toCode(Format.IMAX));
        assertEquals(3, FormatCodes.toCode(Format.IMAX));
    }

    @Test
    void everyFormatSurvivesRoundTripThroughTheMapper() {
        for (Format format : Format.values()) {
            assertEquals(format, FormatCodes.fromCode(FormatCodes.toCode(format)));
        }
    }
}

File: csharp/tests/Training.Workshop.Tests/M6/S03TypeCode/S03SolutionTest.cs
using Training.Workshop.M6.S03TypeCode.Step3;

namespace Training.Workshop.Tests.M6.S03TypeCode;

/// <summary>Granica trwałości: w bazie zostaje stabilny kod, nie wartość liczbowa enuma ani nazwa stałej.</summary>
public sealed class S03SolutionTest
{
    [Fact]
    public void PersistentCodeIsNotTheOrdinal()
    {
        Assert.NotEqual((int)Format.Imax, FormatCodes.ToCode(Format.Imax));
        Assert.Equal(3, FormatCodes.ToCode(Format.Imax));
    }

    [Fact]
    public void EveryFormatSurvivesRoundTripThroughTheMapper()
    {
        foreach (var format in Enum.GetValues<Format>())
        {
            Assert.Equal(format, FormatCodes.FromCode(FormatCodes.ToCode(format)));
        }
    }
}
